apps/os/backend/agent/iterate-agent.ts

```diff
@@ -384,13 +466,51 @@ export class IterateAgent<Slices extends readonly AgentCoreSlice[] = CoreAgentSl
   // Instance-level MCP manager cache and connection queues to avoid sharing across DOs
   protected mcpManagerCache: MCPManagerCache;
   protected mcpConnectionQueues: MCPConnectionQueues;
+  _isInitialized = false;
+
+  // This runs between the synchronous durable object constructor and the asynchronous onStart of the agents SDK
+  // It also performs the PartyKit set-name fetch internally to trigger onStart.
+  async initIterateAgent(params: AgentInitParams) {
+    if (this._isInitialized) return; // no-op if already initialised in this DO lifecycle
 
-  // This gets run between the synchronous durable object constructor and the asynchronous onStart method of the agents SDK
-  async initAfterConstructorBeforeOnStart(params: AgentInitParams) {
     this._databaseRecord = params.record;
     this._estate = params.estate;
     this._organization = params.organization;
     this._iterateConfig = params.iterateConfig;
+
+    // Store init params so onAlarm re-hydration can re-initialise without DB calls
+    await this.ctx.storage.put("agent-init-params", params);
```

Comment:
use the SyncKV, the old one is considered legacy now

```suggestion
    this.ctx.storage.kv.put("agent-init-params", params);
```